### PR TITLE
Rewrite logging

### LINKED ISSUE #743: Configurable logging colors
A user should be able to set the colors of the logging via the configuration. The current colors can be used as default.

I marked this as easy, but it isn't easy for a complete rust beginner, IMO.

### LINKED ISSUE #1012: Rewrite the logging backend
A rewrite of the logging backend should be done.

I will describe how and what should be implemented in a blog post, though, as it fits quite nicely into one.

### PR DESCRIPTION
**Pull request by matthiasbeyer**
As planned in [this blog post](https://imag-pim.org/blog/2017/08/27/planning-a-log-functionality-refactoring-for-imag/) I'm rewriting the logging backend here.

Closes #1012 
Closes #743

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
It cannot find keys in the config, and I'm not sure why that is. All my tests succeeded, even adding a test to the toml-query crate with the config file as test input, also a quick test with toml 0.4.4 and 0.4.5 succeeded, only in imag the keys are not found.

Maybe a bit of sleep will get my head cleared for another try. Tomorrow then.

**Comment by matthiasbeyer** [comment]
@mario-kr if you want to reproduce:

```bash
$ cargo build --manifest-path ./imag-store/Cargo.toml
$ mkdir /tmp/store
$ ./target/debug/imag-store --rtp /tmp/ --config imagrc.toml create -p test entry -c foo
```

prints `ERROR[   1]: Global config 'imag.logging.level' missing`

The corrosponding code which does not find what is needed is in [here](https://github.com/matthiasbeyer/imag/pull/1023/files#diff-c85565c2213a9913c463595ef4f872d6R188).

File: libimagrt/src/logger.rs [hunk 1]
```diff
@@ -19,115 +19,475 @@
...
+    -> Result<LogLevel>
+{
+    match config {
+        Some(cfg) => match cfg
+            .read("imag.logging.level")
+            .map_err_into(EK::ConfigReadError)
+            {
+                Ok(Some(&Value::String(ref s))) => match_log_level_str(s),
+                Ok(Some(_)) => Err(EK::ConfigTypeError.into_error()),
+                Ok(None)    => Err(EK::GlobalLogLevelConfigMissing.into_error()),
```
Comment (mario-kr) [inline libimagrt/src/logger.rs]: This seems to be the error which causes the message you mentioned. Under which circumstances does `cfg.read` return `Ok(None)`?

**Comment by matthiasbeyer** [comment]
As said before, I put this config into a toml-query test:

http://git.imag-pim.org/toml-query-matthias/diff/?h=add-large-read-test&id=291fc95bb19cc8215079cdaf9d121af3162d3f0b&id2=master

Which succeeds.

**Comment by matthiasbeyer** [comment]
The error was that we looked at the wrong place. The error came from another code part.

Now we have the issue that handlebars segfaults.

**Comment by matthiasbeyer** [comment]
The rewrite of the logging functionality is complete and I already tested it with `imag-store`. I guess we can merge this, though I'd like to give @mario-kr the opportunity to play around. I will merge on Sunday, if I have no reports from you, @mario-kr ! :-)

**Comment by matthiasbeyer** [comment]
I guess I'll wait until the "things as traits" branch is merged to master and then rebase this one. Also, there was just a release of handlebars (0.29.1) which fixed the issue documented in this branch. We could update before merging this.

**Comment by matthiasbeyer** [comment]
**Note to self:** Rebase before merging, add to changelog!

**Pull request merged**